### DISCUSSION TITLE
Not sure I understand the output when I use -n=12

### DISCUSSION THREAD

**Discussion by stuaxo** [pytest-dev/pytest-xdist#803] [General]
Why, when I do `pytest -n=12` does it look like I have so many workers ?
I am I misunderstanding something, I thought it would be 12 worker processes.
(I have 8 threads on the CPU, is there something I'm missing?)

```
pytest -n=12
========================== test session starts ==========================
platform linux -- Python 3.8.8, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
Test order randomisation NOT enabled. Enable with --random-order or --random-order-bucket=<bucket_type>
django: settings: settings.test (from ini)
rootdir: /home/stu/projects/work/dit/tamato, configfile: pyproject.toml
plugins: celery-0.0.0, random-order-1.0.4, xdist-2.3.0, allure-pytest-bdd-2.8.33, Faker-8.10.1, responses-0.5.0, django-4.1.0, forked-1.3.0, bdd-4.0.2
gw0 I / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / ggw0 C / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / ggw0 C / gw1 C / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / ggw0 C / gw1 C / gw2 C / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 C / gw5 I / gw6 I / gw7 I / gw8 I / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 C / gw5 C / gw6 I / gw7 I / gw8 I / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 C / gw5 C / gw6 C / gw7 I / gw8 I / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 C / gw5 C / gw6 C / gw7 C / gw8 I / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 C / gw5 C / gw6 C / gw7 C / gw8 C / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 C / gw5 C / gw6 C / gw7 C / gw8 C / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 C / gw5 C / gw6 C / gw7 C / gw8 C / ggw0 C / gw1 C / gw2 C / gw3 C / gw4 C / gw5 gw0 [1814] / gw1 [1814] / gw2 [1814] / gw3 [1814] / gw4 [1814] / gw5 [1814] / gw6 [1814] / gw7 [1814] / gw8 [1814] / gw9 [1814] / gw10 [1814] / gw11 [1814]
```

**Comment by nicoddemus** [comment]
You should have `gw0-gw11` with `-n12`, and that's what I see here... you see it twice perhaps because it shows once for collection, then it shows again with the collected tests when it completes the collection.

**Comment by stuaxo** [comment]
The output looks a little different between say -n9, and -n12.   Have a look at the line just before `scheduling tests via LoadScheduling` for these two runs:

-n9
![image](https://user-images.githubusercontent.com/179677/127207508-ced02be2-a1c1-42b8-b567-eceb2c619d8c.png)

-n12
![image](https://user-images.githubusercontent.com/179677/127207286-530351e3-fba9-433d-ab2a-9b7655ee2e78.png)

**Comment by nicoddemus** [comment]
Hmm seems like it is missing some newlines in there? Some output lines look a bit jammed or something.

**Comment by stuaxo** [comment]
That would make sense, maybe there is an issue output getting buffered (or not) meaning it's not erasing to the beginning of the line.

I could make a video of it if that helps ?

**Comment by nicoddemus** [comment]
> I could make a video of it if that helps ?

I don't think that's necessary, but thanks!

**Comment by TRANTANKHOA** [comment]
Can I also ask in this print out `gw0 [21] / gw1 [21] / gw2 [21] / gw3 [21] / gw4 [21] / gw5 [21] / gw6 [21] / gw7 [21] / gw8 [21] / gw9 [21]` that I get from running 10 workers, what does [21] mean?

**Comment by amyXia1994** [comment]
> Can I also ask in this print out `gw0 [21] / gw1 [21] / gw2 [21] / gw3 [21] / gw4 [21] / gw5 [21] / gw6 [21] / gw7 [21] / gw8 [21] / gw9 [21]` that I get from running 10 workers, what does [21] mean?

I think the 21 means that each worker has collected 21 tests, and that also means that you have totally 21 test cases
